esi/cloud/_lease.py

```diff
@@ -0,0 +1,59 @@
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+import types  # noqa
+
+from esi.lease.v1._proxy import Proxy
+
+
+class LeaseCloudMixin:
+    lease: Proxy
+
+    def __init__(self):
+        super(LeaseCloudMixin, self).__init__()
+        self.cache_enabled = False
+
+    def list_offers(self):
+        """Return a list of all offers."""
+        return list(self.lease.offers(details=True))
+
+    def create_offer(self, resource_id, node_type, **kwargs):
+        """Create an offer"""
+        return self.lease.create_offer(resource_id=resource_id,
+                                       node_type=node_type,
+                                       **kwargs)
+
+    def delete_offer(self, offer):
+        """Delete an offer"""
+        return self.lease.delete_offer(offer)
+
+    def claim_offer(self, offer):
+        """Claim an offer"""
+        return self.lease.claim_offer(offer)
```

Comment:
Would you pass in `new_lease` and/or `**attrs` here?

Reply:
Now I read openstacksdk code again, functions in openstack/cloud/_lease.py belong to Connection class. For example, claim_offer() in esi/cloud/_lease.py is called in this way: connection.Connection.claim_offer(). And for claim_offer() in esi/lease/v1/_proxy.py,  we call it by connection.Connection.lease.v1.claim_offer(). These two are different levels of interface. I saw that baremetal has various APIs on connection.Connection.baremetal and connection.Connection level. I've update the functions in _lease.py to be consistent with those in _proxy.py and added the unit test for them.